sceneview/src/main/java/io/github/sceneview/node/Node.kt

```diff
@@ -469,35 +466,43 @@ open class Node : NodeParent, TransformProvider, SceneLifecycleObserver {
     override fun onFrame(frameTime: FrameTime) {
         // Smooth value compare
         val lerpFactor = clamp((frameTime.intervalSeconds * smoothSpeed).toFloat(), 0.0f, 1.0f)
-        smoothPosition?.takeIf { it != position }?.let { desiredPosition ->
-            position = lerp(position, desiredPosition, lerpFactor).takeIf {
-                distance(position, it) > 0.00001f
-            } ?: desiredPosition
-
-            if (position == desiredPosition) {
-                smoothPosition = null
+        smoothTransform?.let { translation(it) }?.position?.takeIf { it != position }
+            ?.let { desiredPosition ->
+                position = lerp(position, desiredPosition, lerpFactor).takeIf {
+                    distance(position, it) > 0.00001f
+                } ?: desiredPosition
             }
-        }
-        smoothQuaternion?.takeIf { it != quaternion }?.let { desiredQuaternion ->
-            quaternion = slerp(quaternion, normalize(desiredQuaternion), lerpFactor).takeIf {
-                angle(quaternion, desiredQuaternion) > 0.00001f
-            } ?: desiredQuaternion
-            if (quaternion == desiredQuaternion) {
-                smoothQuaternion = null
+        smoothTransform?.let { rotation(it).toQuaternion() }?.takeIf { it != quaternion }
+            ?.let { desiredQuaternion ->
+                quaternion = slerp(quaternion, normalize(desiredQuaternion), lerpFactor).takeIf {
+                    angle(quaternion, desiredQuaternion) > 0.00001f
+                } ?: desiredQuaternion
             }
+        smoothTransform?.let { scale(it) }?.scale?.takeIf { it != scale }?.let { desiredScale ->
+            scale = lerp(scale, desiredScale, lerpFactor).takeIf {
+                distance(scale, it) > 0.00001f
+            } ?: desiredScale
+        }
+
+        if (smoothTransform == transform) {
+            smoothTransform = null
         }
+        updateTransform(worldTransform)
```

Comment:
We should be able to completely remove this method when we use the Filament pick functionality for interacting with the `ViewNode`.